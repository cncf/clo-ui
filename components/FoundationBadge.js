module.exports = require('../dist/cjs/components/FoundationBadge.js');
