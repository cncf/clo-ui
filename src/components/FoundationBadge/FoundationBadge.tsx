import { isNull, isUndefined } from 'lodash';
import React from 'react';

import styles from './FoundationBadge.module.css';

export enum Foundation {
  cdf = 'cdf',
  cncf = 'cncf',
  lfaidata = 'lfaidata',
}

export const FOUNDATIONS = {
  [Foundation.cdf]: {
    name: 'CDF',
  },
  [Foundation.cncf]: {
    name: 'CNCF',
  },
  [Foundation.lfaidata]: {
    name: 'LF AI & Data',
  },
};

export interface IFoundationBadgeProps {
  foundation?: Foundation | null;
  className?: string;
  onClick?: () => void;
}

export const FoundationBadge: React.FC<IFoundationBadgeProps> = (props: IFoundationBadgeProps) => {
  if (isUndefined(props.foundation) || isNull(props.foundation)) return null;

  const foundationData = FOUNDATIONS[props.foundation];

  return (
    <>
      {isUndefined(props.onClick) ? (
        <div
          data-testid="foundation-badge"
          className={`badge text-light extraLightText rounded-0 position-relative ${styles.badge} ${props.className}`}
        >
          <div className="d-flex flex-row align-items-center text-uppercase">{foundationData.name}</div>
        </div>
      ) : (
        <button
          type="button"
          data-testid="foundation-badge"
          className={`badge text-light extraLightText rounded-0 position-relative ${styles.badge} ${props.className}`}
          onClick={props.onClick}
        >
          <div className="d-flex flex-row align-items-center text-uppercase">{foundationData.name}</div>
        </button>
      )}
    </>
  );
};
