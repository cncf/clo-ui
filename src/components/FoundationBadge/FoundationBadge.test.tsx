import { render, screen } from '@testing-library/react';

import { Foundation, FoundationBadge } from './FoundationBadge';

const defaultProps = {
  foundation: Foundation.cncf,
};

describe('FoundationBadge', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<FoundationBadge {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders proper content', () => {
    render(<FoundationBadge {...defaultProps} />);

    expect(screen.getByText('CNCF')).toBeInTheDocument();
    expect(screen.getByTestId('foundation-badge')).toBeInTheDocument();
  });
});
