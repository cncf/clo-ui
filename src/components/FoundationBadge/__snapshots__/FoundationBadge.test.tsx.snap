// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FoundationBadge creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="badge text-light extraLightText rounded-0 position-relative badge undefined"
    data-testid="foundation-badge"
  >
    <div
      class="d-flex flex-row align-items-center text-uppercase"
    >
      CNCF
    </div>
  </div>
</DocumentFragment>
`;
